### PR TITLE
[NCL-8679] add filter archive entries by repo

### PR DESCRIPTION
**Pull request by michalovjan**
Also I've moved the `repository-driver.ignored-repo-patterns` key to `repository-driver.ignored-repo-patterns.promotion` so that it is similar to `repository-driver.ignored-path-patterns.*` now that we have separate entries for archive requests.

### CODE REVIEW

File: src/main/java/org/jboss/pnc/repositorydriver/artifactfilter/ArtifactFilterArchive.java [hunk 1]
```diff
@@ -43,17 +45,24 @@ public class ArtifactFilterArchive implements ArtifactFilter {
...
+                .orElse(new PatternsList(Collections.emptyList()));
     }
 
     @Override
     public boolean accepts(TrackedContentEntryDTO artifact) {
         String path = artifact.getPath();
         StoreKey storeKey = artifact.getStoreKey();
-        return !ignoreContent(ignoredPathPatterns, storeKey.getPackageType(), path);
+        return !ignoreContent(ignoredPathPatterns, storeKey.getPackageType(), path)
+                && !ignoredRepoPatterns.matchesOne(storeKey.toString());
```
Comment (pkocandr) [inline src/main/java/org/jboss/pnc/repositorydriver/artifactfilter/ArtifactFilterArchive.java:65]: Why is this not part of the ignoreContent method?

**Review by pkocandr — CHANGES_REQUESTED** [review]

File: src/main/java/org/jboss/pnc/repositorydriver/artifactfilter/ArtifactFilterArchive.java (hunk 1 above)
Comment (pkocandr) [reply src/main/java/org/jboss/pnc/repositorydriver/artifactfilter/ArtifactFilterArchive.java:65]: Hmm, re-thinking this - if we consider only artifacts as the content, then it is probably correct to separate the repo check from the content check. :-)

File: src/main/java/org/jboss/pnc/repositorydriver/Configuration.java [hunk 2]
```diff
@@ -72,20 +72,23 @@ public class Configuration {
...
-    @ConfigProperty(name = "repository-driver.ignored-repo-patterns")
-    Optional<List<String>> ignoredRepoPatterns;
+    @ConfigProperty(name = "repository-driver.ignored-repo-patterns.promotion")
+    Optional<List<String>> ignoredRepoPatternsPromotion;
 
     @ConfigProperty(name = "repository-driver.ignored-path-patterns.archive.maven")
     Optional<List<String>> ignoredPathPatternsArchiveMaven;
 
+    @ConfigProperty(name = "repository-driver.ignored-repo-patterns.archive")
+    Optional<List<String>> ignoredRepoPatternsArchive;
```
[the remark this answers is not in the document]
Comment (michalovjan) [reply src/main/java/org/jboss/pnc/repositorydriver/Configuration.java:90]: Will do!

**Review by pkocandr — APPROVED** [review]
Looks good now (except for the one extra line because of which the jenkins build fails :-) )

**Pull request merged**